tests/trainer/connectors/test_data_connector.py

```diff
@@ -71,6 +74,115 @@ def test_dataloader_source_request_from_module():
     module.foo.assert_called_once()
 
 
+@pytest.mark.parametrize(
+    "hook_name", ("on_before_batch_transfer", "transfer_batch_to_device", "on_after_batch_transfer")
+)
+@mock.patch("pytorch_lightning.strategies.strategy.Strategy.lightning_module", new_callable=PropertyMock)
+def test_shared_datahook_call(module_mock, hook_name):
```

Comment:
I think it would be simpler and cleaner to unittest the dataclass itself.

There's many other tests that already check that the hooks get called.

Reply:
done!